### PR TITLE
HDDS-4797: Fix findbugs issues after HDDS-2195

### PR DESCRIPTION
**Pull request by GlenGeng**
## What changes were proposed in this pull request?

After spotbugs is enabled with HDDS-2195, there are additional issues reported by findbugs. These issues have to fixed.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4797

## How was this patch tested?

CI

### CODE REVIEW

**Comment by amaliujia** [comment]
`org.apache.hadoop.ozone.recon.api.TestContainerEndpoint` failed due to 500 which seems not relevant.
`org.apache.hadoop.ozone.recon.scm.TestReconContainerManager` probably is still related to recon.


But the acceptance tests seem wired:

```
------------------------------------------------------------------------------
List pipelines on unknown host                                        | FAIL |
'Pipeline[ Id: a7efa2fd-f50d-4176-9749-5dde54cbefd6, Nodes: 7bd494ef-c457-4af3-b39b-6336ccda3d1a{ip: 172.26.0.3, host: ozonescripts_datanode_1.ozonescripts_default, ports: [REPLICATION=9886, RATIS=9858, STANDALONE=9859], networkLocation: /default-rack, certSerialId: null, persistedOpState: IN_SERVICE, persistedOpStateExpiryEpochSec: 0}, Type:STAND_ALONE, Factor:ONE, State:CLOSED, leaderId:, CreationTimestamp2021-02-07T07:45:04.624Z]
Pipeline[ Id: ac7c1c4a-adbe-4be8-bd5f-52de37d4119e, Nodes: 7bd494ef-c457-4af3-b39b-6336ccda3d1a{ip: 172.26.0.3, host: ozonescripts_datanode_1.ozonescripts_default, ports: [REPLICATION=9886, RATIS=9858, STANDALONE=9859], networkLocation: /default-rack, certSerialId: null, persistedOpState: IN_SERVICE, persistedOpStateExpiryEpochSec: 0}, Type:RATIS, Factor:ONE, State:OPEN, leaderId:7bd494ef-c457-4af3-b39b-6336ccda3d1a, CreationTimestamp2021-02-07T07:44:41.689Z]' does not contain 'Invalid host name'
------------------------------------------------------------------------------
Pipeline :: Test ozone admin pipeline command                         | FAIL |
8 critical tests, 7 passed, 1 failed
8 tests total, 7 passed, 1 failed
```

**Comment by amaliujia** [comment]
Overall LGTM if we can confirm the `acceptance tests` are not regression.

**Comment by GlenGeng** [comment]
After last two merge of master, we have four failures, one is findBugs, due to the second merge, 
The first merge cause the failure unit test(TestContainerEndpoint, TestReconContainerManager) and misc docker.
The second merge cause the failure of find bugs.

The fix is aims to the find bugs.

**Comment by GlenGeng** [comment]
> `org.apache.hadoop.ozone.recon.api.TestContainerEndpoint` failed due to 500 error which seems not relevant.
> `org.apache.hadoop.ozone.recon.scm.TestReconContainerManager` probably is still related to recon.
> 
> But the acceptance tests seem wired:
> 
> ```
> ------------------------------------------------------------------------------
> List pipelines on unknown host                                        | FAIL |
> 'Pipeline[ Id: a7efa2fd-f50d-4176-9749-5dde54cbefd6, Nodes: 7bd494ef-c457-4af3-b39b-6336ccda3d1a{ip: 172.26.0.3, host: ozonescripts_datanode_1.ozonescripts_default, ports: [REPLICATION=9886, RATIS=9858, STANDALONE=9859], networkLocation: /default-rack, certSerialId: null, persistedOpState: IN_SERVICE, persistedOpStateExpiryEpochSec: 0}, Type:STAND_ALONE, Factor:ONE, State:CLOSED, leaderId:, CreationTimestamp2021-02-07T07:45:04.624Z]
> Pipeline[ Id: ac7c1c4a-adbe-4be8-bd5f-52de37d4119e, Nodes: 7bd494ef-c457-4af3-b39b-6336ccda3d1a{ip: 172.26.0.3, host: ozonescripts_datanode_1.ozonescripts_default, ports: [REPLICATION=9886, RATIS=9858, STANDALONE=9859], networkLocation: /default-rack, certSerialId: null, persistedOpState: IN_SERVICE, persistedOpStateExpiryEpochSec: 0}, Type:RATIS, Factor:ONE, State:OPEN, leaderId:7bd494ef-c457-4af3-b39b-6336ccda3d1a, CreationTimestamp2021-02-07T07:44:41.689Z]' does not contain 'Invalid host name'
> ------------------------------------------------------------------------------
> Pipeline :: Test ozone admin pipeline command                         | FAIL |
> 8 critical tests, 7 passed, 1 failed
> 8 tests total, 7 passed, 1 failed
> ```

@amaliujia Thanks for the review! The issues mentioned above is due to the merge of recon due to the first master merge.

File: hadoop-hdds/server-scm/dev-support/findbugsExcludeFile.xml
Comment (GlenGeng) [inline hadoop-hdds/server-scm/dev-support/findbugsExcludeFile.xml:51]: This is the same handling way of `org.apache.hadoop.hdds.scm.metadata.TestOMTransactionInfoCodec`. Just keep consistent here.

**Review by bshashikant — APPROVED** [review]
Thanks @GlenGeng for the efforts.

**Comment by amaliujia** [comment]
Thanks @GlenGeng  for your fix!

**Pull request merged**
